### PR TITLE
[MRG] Add multi-simulation support

### PR DESCRIPTION
**Pull request by chenghuzi**
Users now can run multiple simulations and compare them. Also simplified some layout init functions

### CODE REVIEW

File: hnn_core/gui/gui.py [hunk 1]
```diff
@@ -300,32 +251,24 @@ def _initialize_viz_window(simulation_data, analysis_config, init=False):
...
-        viz_grid[layout_option] = grid
-        display(grid)
+    with viz_window:
+        viz_grid[layout_option] = _init_viz_layout(
+            simulation_data,
+            plot_options,
+            previous_plot_types,
+            analysis_config,
+            layout,
+            layout_option,
```
Comment (jasmainak) [inline hnn_core/gui/gui.py:269]: use keyword arguments to be clearer when there are many parameters to the function ... is it `layout` or `style`? Seems like you call it `style` inside the function

Comment (chenghuzi) [reply hnn_core/gui/gui.py:269]: Yeah there's some inconsistency... will fix later. Now I need to fix the GUI test first 😂

**Comment by chenghuzi** [comment]
Now it looks like this: 

image

File: hnn_core/gui/gui.py (hunk 1 above)
Comment (chenghuzi) [reply hnn_core/gui/gui.py:265]: Fixed in commit 0eb29c7a3118f0b2302abed83c3a2e567c4f15af

File: hnn_core/gui/gui.py [hunk 2]
```diff
@@ -43,36 +45,51 @@
...
+        A dict of simulation data
+    analysis_config: dict
+        A dict of visualization configs
+    plot_key: str
+        A string used to identify the plot and data to update.
+    """
     # Make sure that visualization does not change the original data
-    dpls_copied = copy.deepcopy(simulation_data['dpls'])
-    net_copied = copy.deepcopy(simulation_data['net'])
+    _plot_out = analysis_config['plot_outputs'][plot_key]
```
Comment (jasmainak) [inline hnn_core/gui/gui.py:61]: usually leading underscore in the variable name means that it is "private" or for internal use. Is that what you mean here? If not, maybe worth just removing the underscores.

**Comment by jasmainak** [comment]
I'll go ahead and merge, so you can iterate with the new design @chenghuzi . In the future, try to avoid the leading underscores unless you really mean it's a private function or variable.

**Comment by jasmainak** [comment]
Thanks @chenghuzi !

File: hnn_core/gui/gui.py (hunk 2 above)
Comment (chenghuzi) [reply hnn_core/gui/gui.py:61]: You're right I should remove the underscore there as it's safe to do so as it's already in a function. Will clean these sort of stuff later

**Pull request merged**